src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -114,6 +114,11 @@ public function onKernelResponse(FilterResponseEvent $event)
             return;
         }
 
+        if ($this->registered && null !== $this->dispatcher) {
```

Comment:
As the `onKernelResponse` method is only called when we have a dispatcher, and because in this case, registered is always, true, it looks like the if is not needed at all.